packages/donut-monster/test/index.test.js

```diff
@@ -0,0 +1,77 @@
+import getPort from 'get-port'
+import socketIoClient from 'socket.io-client'
+import test from 'ava'
+import { getRandomDonuts, messages } from 'donut-common'
+import { promisify } from 'util'
+
+import io from '../src/index.js'
+
+const getClient = () => socketIoClient(`http://localhost:${io.httpServer.address().port}`)
+
+test.before(async () => {
+  const port = await getPort()
+  return promisify(io.httpServer.listen.bind(io.httpServer))(port)
```

Comment:
Undocumented [`httpServer` property](https://github.com/socketio/socket.io/blob/1c108a35e499579f978908bac3fb47122ed77ee4/lib/index.js#L279-L280) on the SocketServer instance. This happens when manually wiring the `http.Server` on socket.io server construction, as we're doing.